--- a/index.mjs
+++ b/index.mjs
@@ -192,16 +192,18 @@
               })
             }
           },
-          federation({
-            name,
-            exposes: { '.': './src/index.ts' },
-            filename: `${remoteEntryName}${isProduction ? '-[hash]' : ''}${remoteEntryExt}`,
-            shared: Object.fromEntries(
-              externalModules.map((pkg) => [pkg, { singleton: true, import: false }])
-            ),
-            manifest: false,
-            dts: false
-          }),
+          ...(!isTesting
+            ? federation({
+                name,
+                exposes: { '.': './src/index.ts' },
+                filename: `${remoteEntryName}${isProduction ? '-[hash]' : ''}${remoteEntryExt}`,
+                shared: Object.fromEntries(
+                  externalModules.map((pkg) => [pkg, { singleton: true, import: false }])
+                ),
+                manifest: false,
+                dts: false
+              })
+            : []),
           tailwindcss(),
           manifestPlugin(),
           federationRegistrationClient({
